File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program135.cpp
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept one number from user and toggle contents of first and last nibble of the number.
//		 Return modified number
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

typedef unsigned int UINT;

class Solution
{
	public:
		UINT ToggleBit(UINT iValue)
		{
			UINT iMask = 0xF000000F;
			UINT iResult = 0;

			iResult = iValue ^ iMask;

			return iResult;
		}
};

int main()
{
	UINT iValue=0,uRet=0;

	cout<<"Enter number: ";
	cin>>iValue;

	Solution *sObj = new Solution();
	uRet = sObj->ToggleBit(iValue);

	cout<<"Toggled Nibble is: "<<uRet<<"\n";
	return 0;
}

/*

OUTPUT:

Enter number: 10
Toggled Nibble is: 4026531845


Enter number: 4026531845
Toggled Nibble is: 10

*/

File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program135.java
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept one number from user and toggle contents of first and last nibble of the number.
//		 Return modified number
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


import java.util.Scanner;

class Solution
{
	public int iMask,iNo;
	public int iResult;

	Solution(int iNo)
	{
		this.iNo = iNo;
		this.iMask = 0xF000000F;
		this.iResult = 0;
	}

	int ToggleBit()
	{
		iResult = iNo ^ iMask;
		return iResult;
	}	
}

class Program
{
	public static void main(String[] args)
	{
		int iRet=0;

		Scanner sObj = new Scanner(System.in);
		
		System.out.println("Enter number: ");
		int iValue = sObj.nextInt(); 

		Solution sObj1 = new Solution(iValue);
		iRet = sObj1.ToggleBit();

		System.out.println("Modified bit is:"+iRet);
	}	
}
/*
OUTPUT:

*/
















